tabular/src/autogluon/tabular/predictor/predictor.py

```diff
@@ -904,10 +930,27 @@ def fit_extra(self, hyperparameters, time_limit=None, base_model_names=None, **k
         ag_args_fit = kwargs['ag_args_fit']
         ag_args_ensemble = kwargs['ag_args_ensemble']
         excluded_model_types = kwargs['excluded_model_types']
+        pseudo_data = kwargs.get('pseudo_data', None)
+
+        if pseudo_data is not None:
+            if self.label not in pseudo_data.columns:
+                raise ValueError('\'pseudo_data\' does not contain the labeled column.')
+
+            if self.sample_weight is not None:
+                raise ValueError('Applying \'sample_weight\' while calling \'fit_pseudolabel\' is not supported')
+
+            X_pseudo = pseudo_data.drop(columns=[self.label])
+            y_pseudo = pseudo_data[self.label]
+            X_pseudo = self._learner.transform_features(X_pseudo)
+            y_pseudo = self._learner.label_cleaner.transform(y_pseudo)
```

Comment:
Crash in unit test for pre-labeled data for CoverType:

This line of code changes int to float for CoverType because a class in test labels was not in train labels, which converts it to NaN, causing models to crash downstream.

Add a check here if `y_pseudo.unique()` contains np.nan, if so raise exception.

Reply:
added